File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/Fragments/BaseFragment.java
package kore.botssdk.audiocodes.webrtcclient.Fragments;


import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import kore.botssdk.R;
import kore.botssdk.audiocodes.webrtcclient.Activities.SettingsActivity;
import kore.botssdk.audiocodes.webrtcclient.Login.LogoutManager;

public class BaseFragment extends Fragment {
    public Handler handler;

    private final String TAG = "BaseFragment";

    private static final int SETTING_MENU = 100;
    private static final int EXIT_MENU = 101;

    //public List<OptionsMenuObject> optionsMenuObjectList = new ArrayList<OptionsMenuObject>();

    public void onCreate(@Nullable Bundle savedInstanceState) {

        //optionsMenuObjectList = new ArrayList<OptionsMenuObject>();
        super.onCreate(savedInstanceState);
        setMenuVisibility(true);
        setHasOptionsMenu(true);
        handler = new Handler();

    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        // Clear current contents
        menu.clear();
        menu.add(Menu.NONE, SETTING_MENU, Menu.NONE, getString(R.string.menu_option_settings));
        menu.add(Menu.NONE, EXIT_MENU, Menu.NONE, getString(R.string.menu_option_exit));
        super.onCreateOptionsMenu(menu, inflater);
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case SETTING_MENU:
                Intent intent = new Intent(BaseFragment.this.getContext(), SettingsActivity.class);
                startActivity(intent);
                //finish();
                return true;
            case EXIT_MENU:
                LogoutManager.closeApplication();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/fragments/BaseFragment.kt
package com.kore.ui.audiocodes.webrtcclient.fragments

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import androidx.core.view.MenuProvider
import androidx.fragment.app.Fragment
import androidx.lifecycle.Lifecycle
import com.kore.ui.R
import com.kore.ui.audiocodes.webrtcclient.activities.SettingsActivity
import com.kore.ui.audiocodes.webrtcclient.general.ACManager
import com.kore.ui.audiocodes.webrtcclient.login.LogoutManager

open class BaseFragment : Fragment() {
    val acManager: ACManager = ACManager.getInstance()
    var handler: Handler = Handler(Looper.getMainLooper())

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        requireActivity().addMenuProvider(object : MenuProvider {
            override fun onCreateMenu(menu: Menu, menuInflater: MenuInflater) {
                menu.clear()
                menuInflater.inflate(R.menu.main_menu, menu)
            }

            override fun onMenuItemSelected(menuItem: MenuItem): Boolean {
                return when (menuItem.itemId) {
                    R.id.action_settings -> {
                        val intent = Intent(this@BaseFragment.context, SettingsActivity::class.java)
                        startActivity(intent)
                        true
                    }

                    R.id.action_exit -> {
                        LogoutManager.closeApplication(requireContext(), acManager)
                        true
                    }

                    else -> false
                }
            }
        }, viewLifecycleOwner, Lifecycle.State.RESUMED)
    }

    companion object {
        private const val SETTING_MENU = 100
        private const val EXIT_MENU = 101
    }
}